launchdarkly/resource_launchdarkly_view_links_test.go

```diff
@@ -0,0 +1,223 @@
+	project_key = launchdarkly_project.test.key
+	key         = "test-flag-2"
+	name        = "Test Flag 2"
+	variation_type = "boolean"
+}
+
+resource "launchdarkly_feature_flag" "test3" {
+	project_key = launchdarkly_project.test.key
+	key         = "test-flag-3"
+	name        = "Test Flag 3"
+	variation_type = "boolean"
+}
+
+resource "launchdarkly_view_links" "test" {
+	project_key = launchdarkly_project.test.key
+	view_key    = launchdarkly_view.test.key
+	
+	flags = [
+		launchdarkly_feature_flag.test1.key,
+		launchdarkly_feature_flag.test3.key
+	]
+	
+	comment = "Updated test linking flags to view"
+}
+`
+)
+
+func TestAccViewLinks_Create(t *testing.T) {
+	projectKey := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	projectName := "view-links-test-" + projectKey
+	resourceName := "launchdarkly_view_links.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccViewLinksCreate, projectName, projectKey),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckViewLinksExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "project_key", projectKey),
+					resource.TestCheckResourceAttr(resourceName, "view_key", "test-view"),
+					resource.TestCheckResourceAttr(resourceName, "flags.#", "2"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-1"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-2"),
+					resource.TestCheckResourceAttr(resourceName, "comment", "Test linking flags to view"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccViewLinks_Update(t *testing.T) {
+	projectKey := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	projectName := "view-links-test-" + projectKey
+	resourceName := "launchdarkly_view_links.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccViewLinksCreate, projectName, projectKey),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckViewLinksExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "flags.#", "2"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-1"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-2"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(testAccViewLinksUpdate, projectName, projectKey),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckViewLinksExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "flags.#", "2"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-1"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "flags.*", "test-flag-3"),
+					resource.TestCheckResourceAttr(resourceName, "comment", "Updated test linking flags to view"),
+				),
```

Comment:
can you check at the end (do a defer helper function up top) that these flags were unlinked after these were deleted? i'm not actually sure when the teardown function gets called so you may have to add another step with the flag link resource config removed

Reply:
Yep, good shout!